### PR TITLE
Add test coverage for async slicing of labels

### PR DESCRIPTION
**Pull request by kcpevey**
# References and relevant issues
Part of #4795.

# Description
Adds test coverage to check that the layer slicer handles labels layers appropriately.

### CODE REVIEW

**Comment by kcpevey** [comment]
This is my first time working with Labels. Now that I've dipped my toe in the water a bit and attempted an implementation similar to Images, I realize that much of this code (the vast majority) could be removed by allowing `LabelSliceRequest/Response` to inherit from `ImageSliceRequest/Response`. Any opinions on that @potating-potato or @andy-sweet ?

In fact, SO MUCH CODE would be removed that I'm starting to question if I've done this right.

File: napari/layers/labels/labels.py
Comment (kcpevey) [inline napari/layers/labels/labels.py]: All of this will be removed if I inherit from ImageSliceRequest/Response - OR - make a base Request/Response class that is generic so I can pass typing.

**Comment by kcpevey** [comment]
Given discussion on this topic on zulip, I'll be making some major reductions to this PR. Stand by.

File: napari/components/_tests/test_layer_slicer.py
Comment (kcpevey) [inline napari/components/_tests/test_layer_slicer.py]: Required to avoid `ValueError: Input data should be an array-like object, or a sequence of arrays of decreasing size. Got arrays of single shape: (10, 15)` inside of `layer.guess_multiscale()`

**Comment by kcpevey** [comment]
@andy-sweet / @perlman is this in line with what you were thinking?

File: napari/components/_tests/test_layer_slicer.py
Comment (andy-sweet) [reply napari/components/_tests/test_layer_slicer.py]: Normally, passing `multiscale=False` in the layer initializer should be enough to avoid that error. But I'm not against adding this if needed.

Comment (andy-sweet) [reply napari/components/_tests/test_layer_slicer.py]: If this is needed, then maybe just make this a derived property that comes from `self.data.ndim`?

Comment (kcpevey) [reply napari/components/_tests/test_layer_slicer.py]: because of the nonlinear `if` checking, even setting `multiscale=False` doesn't avoid this message. I can change it to a property derived from `self.data.ndim` though.

**Comment by andy-sweet** [comment]
Looks good so far, but as discussed, I think it makes sense to put this one on hold until we have merged an initial integration PR. The PR is already draft, so no need to change anything. Could be interesting to combine this with a PR that enables async slicing/rendering for some of the labels layer specific functionality (e.g. painting).

**Comment by andy-sweet** [comment]
@brisvag : I picked this up while doing some PR grooming. It's very small now and just adds a little more test coverage for async slicing. Would appreciate a review!

File: napari/_tests/utils.py
Comment (brisvag) [inline napari/_tests/utils.py:146]: I guess this protocol is only really used by images and labels? Otherwise this does not really hold true for layers that accept tuples like surface.

Comment (andy-sweet) [reply napari/_tests/utils.py:146]: Yes. This change was needed because `Labels` also needs `ndim` in the test case here, so (at most) `LayerDataProtocol` only supports `Image` right now.

I like the intent of it and it's often useful for prototyping and testing, but would be great to give it a more thought and care. Outside of this PR though ;)

**Review by brisvag — APPROVED** [review]

**Pull request merged**
